src/subject/messages.rs

```diff
@@ -1,54 +1,88 @@
+use crate::database::Executor;
 use actix_web::HttpResponse;
 use async_trait::async_trait;
 use serde::{Deserialize, Serialize};
 
-use super::model::{Subjects, SubjectsError};
 use crate::database::Connection;
 use crate::message::MessageResponder;
 
 #[derive(Deserialize, Serialize)]
 #[serde(tag = "type", content = "payload")]
 pub enum SubjectsMessage {
-    SubjectsQuery(SubjectsQuery),
+    SubjectsQuery(Option<serde_json::Value>),
```

Comment:
Can't we change this to
```suggestion
    SubjectsQuery,
```

Reply:
With your suggested change `{ type: SubjectsQuery, payload: {} }` would result in `400 Bad Request`. With `SubjectsQuery(Option<serde_json::Value>)` we do not need to take care that we omit the payload argument.